addon_service/configured_storage_addon/views.py

```diff
@@ -49,4 +49,5 @@ def get_wb_config(self, request, pk=None):
         addon = self.get_object()
         if addon.external_service.credentials_format is CredentialsFormats.OAUTH2:
             addon.base_account.refresh_oauth_access_token__blocking()
+        self.resource_name = "waterbutler-config"  # for the jsonapi resource type
```

Comment:
oh hey, [it can](https://github.com/encode/django-rest-framework/blob/2da473c8c8e024e80c13a624782f1da6272812da/rest_framework/decorators.py#L163-L166)! but it [requires](https://github.com/encode/django-rest-framework/blob/2da473c8c8e024e80c13a624782f1da6272812da/rest_framework/viewsets.py#L94) those extra kwargs to already exist as attrs on the view -- which seemed fine, added `resource_name: str | None = None` to the viewset class and discovered jsonapi resource type [depends on an AttributeError](https://github.com/django-json-api/django-rest-framework-json-api/blob/027e44cc2219c9517cb6ae8ccf91e6eaf6ef01c8/rest_framework_json_api/utils.py#L52)... could try switching to defining all `resource_name`s on the viewsets instead of on the models (help keep api concerns out of the data layer) but iirc something drf-jsonapi got in the way of that

Reply:
The joys of opinionated frameworks